app/services/products_renderer.rb

```diff
@@ -59,7 +59,12 @@ def distributed_products
   end
 
   def taxon_order
-    if distributor.preferred_shopfront_taxon_order.present?
+    if distributor.preferred_shopfront_product_sorting_method == "by_producer" && distributor.preferred_shopfront_producer_order.present?
```

Comment:
this method is now the wrong name, you should rename it to products_order maybe because it can be taxons or producers now?